pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationDataSource.java

```diff
@@ -82,36 +84,47 @@ default String getHttpHeader(String name) {
 
     /**
      * Check if data from Pulsar protocol are available.
-     * 
+     *
      * @return true if this authentication data contain data from Pulsar protocol
      */
     default boolean hasDataFromCommand() {
         return false;
     }
 
     /**
-     * 
+     *
      * @return authentication data which is stored in a command
      */
     default String getCommandData() {
         return null;
     }
 
+    /**
+     * For mutual authentication, This method use passed in `data` to evaluate and challenge,
```

Comment:
The content is good.
But we should reword this comment, it is not well written IMHO